wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -166,75 +166,57 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
-                FutureUtil
-                  .sequentially(txos)(txo =>
-                    processExistingIncomingTxo(transaction, blockHashOpt, txo))
-                  .map(_.toVector)
-            }
+    def incomingTxoFut: Future[Vector[SpendingInfoDb]] =
+      spendingInfoDAO
+        .findTx(transaction)
+        .flatMap {
+          // no existing elements found
+          case Vector() =>
+            processNewIncomingTx(transaction, blockHashOpt, newTags)
+              .map(_.toVector)
+
+          case txos: Vector[SpendingInfoDb] =>
+            FutureUtil
+              .sequentially(txos)(txo =>
+                processExistingIncomingTxo(transaction, blockHashOpt, txo))
+              .map(_.toVector)
+        }
 
-        val outgoingTxFut: Future[Vector[SpendingInfoDb]] = {
-          for {
-            outputsBeingSpent <-
-              spendingInfoDAO.findOutputsBeingSpent(transaction)
-
-            // unreserved outputs now they are in a block
-            outputsToUse = blockHashOpt match {
-              case Some(_) =>
-                outputsBeingSpent.map { out =>
-                  if (out.state == TxoState.Reserved)
-                    out.copyWithState(TxoState.PendingConfirmationsReceived)
-                  else out
-                }
-              case None =>
-                outputsBeingSpent
+    def outgoingTxFut: Future[Vector[SpendingInfoDb]] = {
+      for {
+        outputsBeingSpent <- spendingInfoDAO.findOutputsBeingSpent(transaction)
+
+        // unreserved outputs now they are in a block
+        outputsToUse = blockHashOpt match {
+          case Some(_) =>
+            outputsBeingSpent.map { out =>
+              if (out.state == TxoState.Reserved)
+                out.copyWithState(TxoState.PendingConfirmationsReceived)
+              else out
             }
-
-            processed <-
-              FutureUtil.sequentially(outputsToUse)(markAsPendingSpent)
-          } yield processed.flatten.toVector
-
+          case None =>
+            outputsBeingSpent
         }
 
-        val aggregateFut =
-          for {
-            incoming <- incomingTxoFut
-            outgoing <- outgoingTxFut
-            _ <-
-              walletCallbacks.executeOnTransactionProcessed(logger, transaction)
-          } yield {
-            ProcessTxResult(incoming.toList, outgoing.toList)
-          }
+        processed <- FutureUtil.sequentially(outputsToUse)(markAsPendingSpent)
+      } yield processed.flatten.toVector
 
-        aggregateFut.failed.foreach { err =>
-          val msg = s"Error when processing transaction=${transaction.txIdBE}"
-          logger.error(msg, err)
-        }
+    }
 
-        aggregateFut
-      }
+    for {
+      incoming <- incomingTxoFut
+      outgoing <- outgoingTxFut
+      _ <- walletCallbacks.executeOnTransactionProcessed(logger, transaction)
     } yield {
-      aggregate
+      ProcessTxResult(incoming.toList, outgoing.toList)
```

Comment:
Since we are touching all of this, let's modify `ProcessTxResult` to be `Vector[]` rather than `List[]`